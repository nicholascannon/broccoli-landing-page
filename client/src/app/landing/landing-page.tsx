import React from 'react';
import { Header } from '../../common/components/header';
import { Footer } from '../../common/components/footer';
import { useShowModal } from '../../common/modals/modal-engine';
import { Button } from '../../common/components/button';

import styles from './landing-page.module.css';

const RequestInviteModal = React.lazy(() => import('./modals/request-invite/request-invite-modal'));

export const LandingPage = () => {
    const showModal = useShowModal();

    const callToAction = () => {
        showModal(RequestInviteModal);
    };

    return <LandingPageView callToAction={callToAction} />;
};

export const LandingPageView = (props: ViewProps) => {
    return (
        <div className={styles.layout}>
            <Header />

            <main className={styles.main}>
                <section className={styles.content}>
                    <h1>A better way to enjoy every day.</h1>
                    <p>Be the first to know when we launch.</p>

                    <Button onClick={props.callToAction}>Request an invite</Button>
                </section>
            </main>

            <Footer />
        </div>
    );
};

type ViewProps = {
    callToAction: () => void;
};
