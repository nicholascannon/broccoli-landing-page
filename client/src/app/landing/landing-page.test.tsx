/* eslint-disable @typescript-eslint/no-empty-function */
import renderer from 'react-test-renderer';
import { LandingPageView } from './landing-page';

describe('LandingPageView', () => {
    test(`
        GIVEN LandingPageView component
        THEN rendering it should match the snapshot
    `, () => {
        const tree = renderer.create(<LandingPageView callToAction={() => {}} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
