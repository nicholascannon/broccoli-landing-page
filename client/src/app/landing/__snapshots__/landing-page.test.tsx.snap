// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
    GIVEN LandingPageView component
    THEN rendering it should match the snapshot
 1`] = `
<div
  className="layout"
>
  <header
    className="header"
  >
    <span
      className="title"
    >
      BROCOLI & CO.
    </span>
  </header>
  <main
    className="main"
  >
    <section
      className="content"
    >
      <h1>
        A better way to enjoy every day.
      </h1>
      <p>
        Be the first to know when we launch.
      </p>
      <button
        className="button undefined "
        onClick={[Function]}
      >
        Request an invite
      </button>
    </section>
  </main>
  <footer
    className="footer"
  >
    <p>
      Made with ♥ in Melbourne.
    </p>
    <p>
      © 2023 Broccoli & Co. All rights reserved.
    </p>
  </footer>
</div>
`;
